### DISCUSSION TITLE
Using emojis and accented characters in executed (ba)sh code blocks

### DISCUSSION THREAD

**Discussion by Enn83** [pawamoy/markdown-exec#37] [Q&A] [answered]
Hi!

I was trying to execute a sh code block, but it contain caracters like `é`, `è`, `à`, etc. and even emojis `💎`, `🐍`, `🔓`, `🔐`, `📦`, etc.
But when I use them:
* Firstly the `é`, `è`, `à` caracters are rendered as `Ã©`, `Ã¨` and `Ã `.
* Secondly, when I add an emoji, all the code block is transformed into an inline code (or no code formatting at all).

For example:
````sh
```bash exec="1" result="ansi"
printf '💎, 🐍, 🔓, 🔐, 📦, 🙃\n'
```

```bash exec="1" result="ansi"
printf 'éèàùôç'
```

```bash exec="1" result="ansi"
printf '\x1b[32mHere it works !\x1b[91m.\n'
printf '\x1b[3;33mHere it works\x1b[23;39m\n'
```
````

... is rendered as...

![image](https://github.com/pawamoy/markdown-exec/assets/46783975/7765a0ba-5d04-410a-9da1-1ac033db7c7d)

Is there a way to display these caracters in an executed code block?

Thanks for your time 😊

-----

***Additional question:*** *Is there any reason there is no support for italic and darker/lighter/etc. color code? For example `\x1b[3m` for italic, `\x1b[91m` for bright red (and so `\x1b[31m` for red)? Perhaps for accessibility reasons?*

**Comment by pawamoy** [comment]
It seems to work fine on my end :slightly_smiling_face: 

![emojis](https://github.com/pawamoy/markdown-exec/assets/3999221/09bad5dd-5817-48b5-b354-4bd140667daf)

Can you check if your file is encoded in UTF8?

**Reply by Enn83** [reply]
I checked, and all my files are encoded in UTF-8.
I will try with a minimal example and see if another plugin interferes with `markdown-exec` rendering 🤔 ...

*Also for my additional question, I found color-code detection is defined in [ `pygments_ansi_color`](https://github.com/chriskuehl/pygments-ansi-color), so I will ask there!*

**Reply by Enn83** [reply]
I tried to make a *minimal reproductible example*, attached below, but I am not able to run code blocks, despite I followed the [documentation](https://pawamoy.github.io/markdown-exec/) 😣 :

[9.5.15-markdown-exec-utf8.zip](https://github.com/pawamoy/markdown-exec/files/14743472/9.5.15-markdown-exec-utf8.zip)

![image](https://github.com/pawamoy/markdown-exec/assets/46783975/81ef4338-b228-463b-82ec-c72c6ca77ae6)

**Reply by pawamoy** [reply]
Ah, sorry, you need to list `pymdownx.superfences` into `markdown_extensions` in `mkdocs.yml`. I opened an issue to improve the situation: https://github.com/pawamoy/markdown-exec/issues/39.

**Reply by Enn83** [reply]
Thanks 😀 !

I added `pymdownx.superfences` in the `pymdownx.superfences` section of my *minimal reproductible example* and now code blocks are rendered, but I have the same issue :

![image](https://github.com/pawamoy/markdown-exec/assets/46783975/af5d084e-0c93-4bcf-a2fc-ebb543960c5d)

Do you have any clue why emojis and other doesn't work on my computer (Windows 11)?

[9.5.15-markdown-exec-utf8.zip](https://github.com/pawamoy/markdown-exec/files/14744720/9.5.15-markdown-exec-utf8.zip)

**Reply by pawamoy** [reply]
I noticed that your files are encoded as UTF-8 with Bom. Try re-encoding them as just UTF-8.

**Reply by Enn83** [reply]
Indeed I tried several encoding (including `UTF-8 with BOM`) but my files were initially in `UTF-8`. I tried again to save them in `UTF-8` and I have the same result 😣 :

![image](https://github.com/pawamoy/markdown-exec/assets/46783975/927eaa31-1731-4b8b-b730-6efbacb989f9)

**Reply by pawamoy** [reply, marked as answer]
Try setting these environment variables on your system:

- `PYTHONUTF8=1`
- `PYTHONIOENCODING=UTF8`

If that still does not work, I'll just blame the annoying operating system and leave it at that :shrug:

**Reply by Enn83** [reply]
It works!

Adding this variable indeed fix the issue:
```sh
export PYTHONUTF8=1
```

![image](https://github.com/pawamoy/markdown-exec/assets/46783975/5bd69af2-c44a-46ea-a766-b929044cd27b)

Thanks for your help ❤️!

**Reply by pawamoy** [reply]
Great :smile: You're welcome :tada:
